buildvertical.targets

```diff
@@ -5,106 +5,82 @@
     <ImportedBuildVerticalTargets>true</ImportedBuildVerticalTargets>
   </PropertyGroup>
 
- <!-- Required for FindBestConfiguration task -->
- <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
+  <!-- Import configuration data model -->
+  <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
 
- <Target Name="AnnotateProjects"
-         BeforeTargets="BuildAllProjects"
-         Condition="!$(MSBuildProjectFullPath.EndsWith('.proj'))">
-   <!-- Clear ProjectWithConfiguration to prevent circular dependency -->
-   <ItemGroup><ProjectWithConfiguration Remove="@(ProjectWithConfiguration)" /></ItemGroup>
-   <MSBuild Targets="AnnotateProjectsWithConfiguration" 
-            Projects="@(Project)"> 
-     <Output TaskParameter="TargetOutputs" 
-             ItemName="ProjectWithConfiguration" /> 
-   </MSBuild> 
-   <ItemGroup>
-     <Project Remove="@(Project)" />
-     <Project Include="@(ProjectWithConfiguration)" />
-   </ItemGroup>
- </Target>
+  <!-- Runs during traversal to select which projects and configurations of those projects to build -->
+  <Target Name="AnnotateAndFilterProjects"
+          BeforeTargets="BuildAllProjects;TestAllProjects"
+          Condition="!$(MSBuildProjectFullPath.EndsWith('.proj'))">
```

Comment:
I'd assume that Eric carried this over from the previous version.  The only ".proj" file in the traversal at the time was build.proj and we didn't want to annotate / trim it.  I'd think that it's now safe to remove that condition given other change recently.